clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -547,34 +580,47 @@ private void updateGroupSubscription(Set<String> topics) {
         return groupAssignment;
     }
 
-    private void adjustAssignment(final Map<TopicPartition, String> ownedPartitions,
-                                  final Map<String, Assignment> assignments) {
-        boolean revocationsNeeded = false;
-        Set<TopicPartition> assignedPartitions = new HashSet<>();
+    /**
+     * Used by COOPERATIVE rebalance protocol only.
+     *
+     * Validate the user assignor returned assignments such that there are no partitions which are going to
+     * be assigned to a different consumer other than its current owner (i.e. who includes it in its owned
+     * partitions list encoded in subscription); also if there are any revoked partitions set the error code to let
+     * everyone retry.
+     */
+    private void validateAssignment(final Map<String, List<TopicPartition>> ownedPartitions,
```

Comment:
nit: if you change the name to `validateCooperativeAssignment`, then you can remove the comment about usage.